### PR TITLE
Add fire event state compiler feature

### PR DESCRIPTION
**Pull request by UtahDave**

### CODE REVIEW

File: salt/state.py [hunk 1]
```diff
@@ -1775,16 +1775,36 @@ def check_requisite(self, low, running, chunks, pre=False):
 
         return status, reqs
 
-    def event(self, chunk_ret, length):
+    def event(self, chunk_ret, length, fire_event_flag=False):
```
Comment (cachedout) [inline salt/state.py]: This should just be ```fire_event```. We already know it's a flag. :]

File: salt/state.py [hunk 2]
```diff
@@ -1775,16 +1775,36 @@ def check_requisite(self, low, running, chunks, pre=False):
...
+        chunk is evaluated an event will be set up to the master with the
+        results.
         '''
-        if not self.opts.get('local') and self.opts.get('state_events', True) and self.opts.get('master_uri'):
-            ret = {'ret': chunk_ret,
-                   'len': length}
-            tag = salt.utils.event.tagify(
-                    [self.jid, 'prog', self.opts['id'], str(chunk_ret['__run_num__'])], 'job'
-                    )
+        if not self.opts.get('local') and (self.opts.get('state_events', True) or fire_event_flag) and self.opts.get('master_uri'):
```
Comment (cachedout) [inline salt/state.py]: These are difficult to read in Python (at least for me). An inline comment would help.

**Comment by cachedout** [comment]
Did you test this against states which do not accept ```**kwargs```? It might not work.

File: salt/state.py (hunk 1 above)
Comment (UtahDave) [inline salt/state.py]: Fixed.

**Comment by thatch45** [comment]
@cachedout it should work, he added it to the runtime keywords

**Pull request merged**